File: MobilneSystemyInformatyczne/ShareWindsurfingEquipment/src/main/java/com/project/ShareWindsurfingEquipment/model/Rental.java
package com.project.ShareWindsurfingEquipment.model;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.math.BigDecimal;
import java.util.Set;
import java.util.Set;

@Entity
@Table
public class Rental {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private BigDecimal totalPrice;

    @Column
    private String dateRental;

    @Column
    private String hourRental;

    @Column
    private String lender;

    @Column
    private String status;

    public String getLender() {
        return this.lender;
    }

    public void setLender(String lender) {
        this.lender = lender;
    }

    public String getStatus() {
        return this.status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    @Column
    private int duration;

    @ManyToOne
    @JoinColumn(name = "user_account_id")
    private UserAccount userAccount;

    @ManyToMany(mappedBy = "rental")
	private Set<Equipment> equipment;

    public Set<Equipment> getEquipments() {
        return this.equipment;
    }

    public void setEquipments(Set<Equipment> equipments) {
        this.equipment = equipments;
    }

    public Rental(BigDecimal totalPrice, String dateRental, String hourRental, int duration, Set<Equipment> equipments, String lender, String status) {
        this.totalPrice = totalPrice;
        this.dateRental = dateRental;
        this.hourRental = hourRental;
        this.duration = duration;
        this.lender = lender;
        this.status = status;
        this.equipment = equipments;
    }


    public Rental(Long id, BigDecimal totalPrice, String dateRental, String hourRental, int duration, UserAccount userAccount, Set<Equipment> equipment, String lender, String status) {
        this.id = id;
        this.totalPrice = totalPrice;
        this.dateRental = dateRental;
        this.hourRental = hourRental;
        this.duration = duration;
        this.userAccount = userAccount;
        this.equipment = equipment;
        this.lender = lender;
        this.status = status;
    }

    public Rental() {
  
    }

    public void addEquipment(Equipment equipment){ 
        this.equipment.add(equipment);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public BigDecimal getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(BigDecimal totalPrice) {
        this.totalPrice = totalPrice;
    }

    public String getDateRental() {
        return dateRental;
    }

    public void setDateRental(String dateRental) {
        this.dateRental = dateRental;
    }

    public int getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }

    public UserAccount getUserAccount() {
        return userAccount;
    }

    public void setUserAccount(UserAccount userAccount) {
        this.userAccount = userAccount;
    }

    public String getHourRental() {
        return this.hourRental;
    }

    public void setHourRental(String hourRental) {
        this.hourRental = hourRental;
    }

    @Override
    public String toString() {
        return "Rental{" +
                "id=" + id +
                ", totalPrice=" + totalPrice +
                ", dateRental=" + dateRental +
                ", duration=" + duration +
                ", userAccount=" + userAccount +
                '}';
    }
}

File: MSI-project/src/app/models/rental.ts
import { UserAccount } from "./UserAccount";

export class Rental {


  constructor(
    private _id: number,
    private _totalPrice: number,
    private _dateRental: string,
    private _hourRental: string,
    private _status: string,
    private _lender: string,
    private _duration: number,
    private _userAccount: UserAccount
  ){}

  public get userAccount(): UserAccount {
    return this._userAccount;
  }
  public set userAccount(value: UserAccount) {
    this._userAccount = value;
  }
  public get duration(): number {
    return this._duration;
  }
  public set duration(value: number) {
    this._duration = value;
  }
  public get status(): string {
    return this._status;
  }
  public set status(value: string) {
    this._status = value;
  }
  public get lender(): string {
    return this._status;
  }
  public set lender(value: string) {
    this._status = value;
  }
  public get dateRental(): string {
    return this._dateRental;
  }
  public set dateRental(value: string) {
    this._dateRental = value;
  }

  public get hourRental(): string {
    return this._hourRental;
  }
  public set hourRental(value: string) {
    this._hourRental = value;
  }

  public get totalPrice(): number {
    return this._totalPrice;
  }
  public set totalPrice(value: number) {
    this._totalPrice = value;
  }
  public get id(): number {
    return this._id;
  }
  public set id(value: number) {
    this._id = value;
  }
}
